File: src/main/java/com/yupi/usercenter/model/UserDTO.java
package com.yupi.usercenter.model;

import com.yupi.usercenter.constant.UserConstant;
import lombok.Data;
import lombok.NonNull;
import org.springframework.lang.Nullable;

import java.util.Date;

/**
 * @see com.yupi.usercenter.model.User
 * @author lipeng
 * @since 2025/5/11 7:54
*/
@Data
public class UserDTO {

    public UserDTO(@NonNull User userPO) {
        this.userId = userPO.getId();
        this.userName = userPO.getUserName();
        this.avatarUrl = userPO.getAvatarUrl();
        this.gender = userPO.getGender() == 0 ? "男" : "女";
        this.email = userPO.getEmail();
        this.phone = userPO.getPhone();
        this.createDatetime = userPO.getCreateDatetime();
        this.userRole = userPO.getUserRole() == 0 ? UserConstant.USER_ROLE_DEFAULT : UserConstant.USER_ROLE_ADMIN;
        this.tags = userPO.getTagJsonList();
        this.state = userPO.getIsValid() == 0 ? "normal" : "invalid";
    }

    @NonNull
    private Long userId;

    @NonNull
    private String userName;

    /**
     * 用户头像URL
     */
    @Nullable
    private String avatarUrl;

    @Nullable
    private String gender;

    /**
     * 手机号
     */
    @Nullable
    private String phone;

    /**
     * 邮箱
     */
    @Nullable
    private String email;

    /**
     * 用户注册时间
     */
    @NonNull
    private Date createDatetime;

    /**
     * 用户角色
     */
    @NonNull
    private String userRole;

    /**
     * 用户状态
    */
    @NonNull
    private String state;

    /**
     * 用户标签-JSON列表
     */
    @Nullable
    private String tags;

    private static final long serialVersionUID = 1L;
}

File: src/main/kotlin/com/yupi/usercenter/model/dto/UserDTO.kt
package com.yupi.usercenter.model.dto

import com.fasterxml.jackson.annotation.JsonFormat
import org.springframework.lang.Nullable
import java.io.Serializable
import java.time.LocalDateTime

/**
 * @author lipeng
 * @see com.yupi.usercenter.model.User
 * @since 2025/5/11 7:54
 */

data class UserDTO(
        @Nullable
        var userId: Long? = null,

        @Nullable
        var userName: String? = null,

        /**
         * 用户头像URL
         */
        @Nullable
        var avatarUrl: String? = null,

        @Nullable
        var gender: String? = null,

        /**
         * 手机号
         */
        @Nullable
        var phone: String? = null,

        /**
         * 邮箱
         */
        @Nullable
        var email: String? = null,

        /**
         * 用户注册时间
         */
        @Nullable
        @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
        var createDatetime: LocalDateTime? = null,

        /**
         * 用户角色
         */
        @Nullable
        var userRole: String? = null,

        /**
         * 用户状态
         */
        @Nullable
        var state: String? = null,
) : Serializable {
    companion object {
        private const val serialVersionUID = 1L
    }
}